File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/CategoryMealsForm.ts
import { StringEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CategoryMealsForm {
    CategoryMealsName: StringEditor;
    Photo: ImageUploadEditor;
    Description: StringEditor;
}

export class CategoryMealsForm extends PrefixedContext {
    static readonly formKey = 'Default.CategoryMeals';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoryMealsForm.init)  {
            CategoryMealsForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;

            initFormType(CategoryMealsForm, [
                'CategoryMealsName', w0,
                'Photo', w1,
                'Description', w0
            ]);
        }
    }
}

File: Restaurant/Restaurant.Web/Modules/Default/CategoryMeals/CategoryMealsForm.cs
using Serenity.ComponentModel;

namespace Restaurant.Default.Forms;

[FormScript("Default.CategoryMeals")]
[BasedOnRow(typeof(CategoryMealsRow), CheckNames = true)]
public class CategoryMealsForm
{
    [Required]
    public string CategoryMealsName { get; set; }
 
    public string Photo { get; set; }
    public string Description { get; set; }
}